### PR TITLE
Deprecate Rhino and Nashorn JavaScript engines (#5515)

### PR DESCRIPTION
**Pull request by jkosternl**

### CODE REVIEW

File: core/src/main/java/nl/nn/adapterframework/senders/JavascriptSender.java [hunk 1]
```diff
@@ -68,7 +69,7 @@ public class JavascriptSender extends SenderSeries {
 	private String fileInput;
 
 	public enum JavaScriptEngines {
-		J2V8(J2V8.class), NASHORN(Nashorn.class), RHINO(Rhino.class);
+		J2V8(J2V8.class), @Deprecated NASHORN(Nashorn.class), @Deprecated RHINO(Rhino.class);
```
Comment (tnleeuw) [inline core/src/main/java/nl/nn/adapterframework/senders/JavascriptSender.java:72]: Can you add `@ConfigurationWarning` on the deprecated elements too? Or will that not be displayed in the ConfigurationWarnings anyway?

File: core/src/main/java/nl/nn/adapterframework/senders/JavascriptSender.java [hunk 2]
```diff
@@ -207,9 +208,12 @@ public void setJsFunctionName(String jsFunctionName) {
 	}
 
 	/**
-	 * the name of the javascript engine to be used
+	 * the name of the JavaScript engine to be used.
+	 * @deprecated Both Nashorn and Rhino are deprecated. Use J2V8 instead.
```
Comment (tnleeuw) [inline core/src/main/java/nl/nn/adapterframework/senders/JavascriptSender.java:212]: Please add that Nashorn and Rhine will be removed in 8.0.

File: core/src/main/java/nl/nn/adapterframework/senders/JavascriptSender.java [hunk 3]
```diff
@@ -207,9 +208,12 @@ public void setJsFunctionName(String jsFunctionName) {
...
 	 * @ff.default J2V8
 	 */
+	@Deprecated
+	@ConfigurationWarning("JavaScript engines Nashorn and Rhino deprecated. Use \"J2V8\" instead")
```
Comment (tnleeuw) [inline core/src/main/java/nl/nn/adapterframework/senders/JavascriptSender.java:216]: Please also add here that Nashorn and Rhine will be deleted in 8.0

**Review by tnleeuw — CHANGES_REQUESTED** [review]

**Comment by tnleeuw** [comment]
Can you please also add on `RhinoPipe` the `@ConfigurationWarning` element that it will be deleted in 8.0 and instead the `JavascriptSender` should be used?
That way anyone still using it will get a more direct warning about this.

File: core/src/main/java/nl/nn/adapterframework/senders/JavascriptSender.java (hunk 2 above)
Comment (tnleeuw) [reply core/src/main/java/nl/nn/adapterframework/senders/JavascriptSender.java:212]: As discussed in standup, not needed.

File: core/src/main/java/nl/nn/adapterframework/senders/JavascriptSender.java (hunk 3 above)
Comment (tnleeuw) [reply core/src/main/java/nl/nn/adapterframework/senders/JavascriptSender.java:216]: As discussed in standup, not needed.

Comment (jkosternl) [reply core/src/main/java/nl/nn/adapterframework/senders/JavascriptSender.java:216]: Thanks

File: core/src/main/java/nl/nn/adapterframework/senders/JavascriptSender.java (hunk 1 above)
Comment (jkosternl) [reply core/src/main/java/nl/nn/adapterframework/senders/JavascriptSender.java:72]: I'll add it anyways.. is faster than checking if it is being used. We'll remove it in 8.0 anyways.

**Review by tnleeuw — APPROVED** [review]

File: core/src/main/java/nl/nn/adapterframework/senders/JavascriptSender.java [hunk 4]
```diff
@@ -67,10 +68,18 @@ public class JavascriptSender extends SenderSeries {
...
+
 	public enum JavaScriptEngines {
-		J2V8(J2V8.class), NASHORN(Nashorn.class), RHINO(Rhino.class);
+		J2V8(J2V8.class),
+
+		@ConfigurationWarning("Engine Nashorn is deprecated. Use J2V8 instead.")
+		@Deprecated NASHORN(Nashorn.class),
+
+		@ConfigurationWarning("Engine Rhino is deprecated. Use J2V8 instead.")
+		@Deprecated RHINO(Rhino.class);
```
Comment (nielsm5) [inline core/src/main/java/nl/nn/adapterframework/senders/JavascriptSender.java:79]:
```suggestion
		@Deprecated NASHORN(Nashorn.class),

		@Deprecated RHINO(Rhino.class);
```
Just the deprecated tag is enough here to 'hide' the element, the ConfigurationWarning should be on the setter element instead.

**Pull request merged**